src/Asset/AssetManager.php

```diff
@@ -258,251 +327,442 @@ public function getBundle($name, $publish = true)
+     * server.
+     *
+     * If the asset is a file, its file modification time will be checked to avoid unnecessary file copying.
+     *
+     * If the asset is a directory, all files and subdirectories under it will be published recursively. Note, in case
+     * $forceCopy is false the method only checks the existence of the target directory to avoid repetitive copying
+     * (which is very expensive).
+     *
+     * By default, when publishing a directory, subdirectories and files whose name starts with a dot "." will NOT be
+     * published. If you want to change this behavior, you may specify the "beforeCopy" option as explained in the
+     * `$options` parameter.
+     *
+     * Note: On rare scenario, a race condition can develop that will lead to a  one-time-manifestation of a
+     * non-critical problem in the creation of the directory that holds the published assets. This problem can be
+     * avoided altogether by 'requesting' in advance all the resources that are supposed to trigger a 'publish()' call,
+     * and doing that in the application deployment phase, before system goes live. See more in the following
+     * discussion: http://code.google.com/p/yii/issues/detail?id=2579
+     *
+     * @param string $path the asset (file or directory) to be published
+     * @param array $options the options to be applied when publishing a directory. The following options are
+     * supported:
+     *
+     * - only: array, list of patterns that the file paths should match if they want to be copied.
+     * - except: array, list of patterns that the files or directories should match if they want to be excluded from
+     *   being copied.
+     * - caseSensitive: boolean, whether patterns specified at "only" or "except" should be case sensitive. Defaults to
+     *   true.
+     * - beforeCopy: callback, a PHP callback that is called before copying each sub-directory or file.
+     *   This overrides {@see beforeCopy} if set.
+     * - afterCopy: callback, a PHP callback that is called after a sub-directory or file is successfully copied. This
+     *   overrides {@seee afterCopy} if set.
+     * - forceCopy: boolean, whether the directory being published should be copied even if it is found in the target
+     *   directory. This option is used only when publishing a directory. This overrides {@see forceCopy} if set.
+     *
+     * @return array the path (directory or file path) and the URL that the asset is published as.
+     * @throws \InvalidArgumentException if the asset to be published does not exist.
      *
-     * @return string|bool
      */
-    protected function resolveAsset($bundle, $asset)
+    public function publish(string $path, array $options = []): array
     {
-        if (isset($this->assetMap[$asset])) {
-            return $this->assetMap[$asset];
+        $path = $this->aliases->get($path);
```

Comment:
We can likely leave alias resolving to the end user of the method i.e. remove ` $path = $this->aliases->get($path);`. If one needs an alias, it's easy get value from it.

Reply:
Yes but is there a reason for it to understand aliases instead accepting paths already resolved?

Reply:
OK. Let's have it for now.